### PR TITLE
EN-3436, add new functionality additional_activity_handler

### PR DESCRIPTION
**Pull request by nitzanpx**
Also needed to add some items to ngx.ctx so it could be used for automation tests

### CODE REVIEW

File: lib/px/pxconfig.lua [hunk 1]
```diff
@@ -4,24 +4,31 @@
...
 local _M = {}
 
 _M.px_enabled = true
 
 -- ##  Configuration Block ##
-
-_M.px_appId = 'PX_APP_ID'
-_M.cookie_secret = 'COOKIE_SECRET'
-_M.auth_token = 'PX_AUTH_TOKEN'
+_M.px_appId = 'PXvRfnOj4y'
```
Comment (yaronpx) [inline lib/px/pxconfig.lua]: You should remove this

File: lib/px/pxconfig.lua [hunk 2]
```diff
@@ -4,24 +4,31 @@
...
 _M.blocking_score = 60
 _M.cookie_encrypted = true
 _M.enable_server_calls = true
 _M.send_page_requested_activity = true
 _M.block_enabled = true
 _M.captcha_enabled = true
 _M.px_debug = true
-_M.sensitive_routes_prefix = {}
-_M.sensitive_routes_suffix = {}
+_M.additional_activity_handler = function(event_type, ngx, details)
```
Comment (yaronpx) [inline lib/px/pxconfig.lua]: Why is this function part of `pxconfig`?

**Comment by yaronpx** [comment]
Like always, you should do something about the indentation of your code

**Comment by nitzanpx** [comment]
@yaronpx please re-review

File: lib/px/utils/pxclient.lua [hunk 3]
```diff
@@ -86,6 +86,11 @@ function M.load(config_file)
         if event_type == 'page_requested' and not px_config.send_page_requested_activity then
             return
         end
+			
+	if px_config.additional_activity_handler ~= nil then
+		px_logger.debug("additional_activity_handler was triggered");
+		px_config.additional_activity_handler(event_type, ngx.ctx, details)
```
Comment (yaronpx) [inline lib/px/utils/pxclient.lua]: You're calling a function that doesn't exist

Comment (nitzanpx) [inline lib/px/utils/pxclient.lua]: Its a function that the client should implement, thats why its additional
see in [pxconfig](https://github.com/PerimeterX/perimeterx-nginx-plugin/pull/50/files/fc5abe8f959277deb5850cc6b8ee9e52164af948#diff-150cb6ed60d368445c00b5b980c38d48R25)

**Comment by nitzanpx** [comment]
Updated readme file, @bendpx / @yaronpx  please approve and merge

**Pull request merged**